intermediate_source/DCP_tutorial.rst

```diff
@@ -0,0 +1,238 @@
+
+        def forward(self, x):
+            return self.net2(self.relu(self.net1(x)))
+
+
+    def setup(rank, world_size):
+        os.environ["MASTER_ADDR"] = "localhost"
+        os.environ["MASTER_PORT"] = "12355 "
+
+        # initialize the process group
+        dist.init_process_group("nccl", rank=rank, world_size=world_size)
+        torch.cuda.set_device(rank)
+
+
+    def cleanup():
+        dist.destroy_process_group()
+
+
+    def run_fsdp_checkpoint_load_example(rank, world_size):
+        print(f"Running basic FSDP checkpoint loading example on rank {rank}.")
+        setup(rank, world_size)
+
+        # create a model and move it to GPU with id rank
+        model = ToyModel().to(rank)
+        model = FSDP(model)
+
+        FSDP.set_state_dict_type(
+            model,
+            StateDictType.SHARDED_STATE_DICT,
+        )
+        state_dict = {
+            "model": model.state_dict(),
+        }
+
+        DCP.load_state_dict(
+            state_dict=state_dict,
+            storage_reader=DCP.FileSystemReader(CHECKPOINT_DIR),
+        )
+        model.load_state_dict(state_dict["model"])
+
+        cleanup()
+
+
+    if __name__ == "__main__":
+        world_size = torch.cuda.device_count()
+        print(f"Running fsdp checkpoint example on {world_size} devices.")
+        mp.spawn(
+            run_fsdp_checkpoint_load_example,
+            args=(world_size,),
+            nprocs=world_size,
+            join=True,
+        )
+
+If you would like to load the saved checkpoint into a non-FSDP wrapped model in a non distributed setup, perhaps for inference, you can also do that with DCP.
+By default, DCP saves and loads a distributed state_dict in Single Program Multiple Data(SPMD) style. To load without a distributed setup, please set ``no_dist`` to ``True`` when loading with DCP.
+
+.. code-block:: python
+    import os
+
+    import torch
+    import torch.distributed.checkpoint as DCP
+    import torch.nn as nn
+
+
+    CHECKPOINT_DIR = "checkpoint"
+
+
+    class ToyModel(nn.Module):
+        def __init__(self):
+            super(ToyModel, self).__init__()
+            self.net1 = nn.Linear(16, 16)
+            self.relu = nn.ReLU()
+            self.net2 = nn.Linear(16, 8)
+
+        def forward(self, x):
+            return self.net2(self.relu(self.net1(x)))
+
+
+    def run_checkpoint_load_example():
+        # create the non FSDP-wrapped toy model
+        model = ToyModel()
+        state_dict = {
+            "model": model.state_dict(),
+        }
+
+        # turn no_dist to be true to load in non-distributed setting
+        DCP.load_state_dict(
+            state_dict=state_dict,
+            storage_reader=DCP.FileSystemReader(CHECKPOINT_DIR),
+            no_dist=True,
+        )
+        model.load_state_dict(state_dict["model"])
+
+    if __name__ == "__main__":
+        print(f"Running basic DCP checkpoint loading example.")
+        run_checkpoint_load_example()
```

Comment:
Add a Conclusion section. 

Conclusion
--------------

In this tutorial, we have learned .....
For more information, see the following: 
* Link1
* Link2
* LInk3